client/src/components/Settings/SettingsContributing.res

```diff
@@ -218,71 +218,104 @@ module UseAssets = {
 // -------------------
 module General = {
   @ppx.deriving(show)
-  type rec t = {
-    showSidebarDebuggerPanel: bool,
-    allowTuples: bool
-  }
+  type rec t = {showSidebarDebuggerPanel: bool}
 
   @ppx.deriving(show)
-  type rec msg =
-    | SetSidebarPanel(bool)
-    | SetTuplesAllowed(bool)
+  type rec msg = SetSidebarPanel(bool)
 
   @ppx.deriving(show)
   type rec intent = unit
 
   let default = {
     showSidebarDebuggerPanel: false,
-    allowTuples: false
   }
 
   let toSaved = (state: t) => {
     open Json.Encode
-    object_(list{
-      ("showSidebarDebuggerPanel", bool(state.showSidebarDebuggerPanel)),
-      ("allowTuples", bool(state.allowTuples))
-    })
+    object_(list{("showSidebarDebuggerPanel", bool(state.showSidebarDebuggerPanel))})
   }
   let fromSaved = (j: Js.Json.t) => {
     open Json.Decode
     {
       showSidebarDebuggerPanel: field("showSidebarDebuggerPanel", bool, j),
+    }
+  }
+
+  let init = () => Tea.Cmd.none
+
+  let update = (_state: t, msg: msg): (t, intent) =>
+    switch msg {
+    | SetSidebarPanel(v) => ({showSidebarDebuggerPanel: v}, ())
+    }
+}
+
+// -------------------
+// Toggle to preview in-progress features
+// -------------------
+module InProgressFeatures = {
+  @ppx.deriving(show)
+  type rec t = {allowTuples: bool}
+
+  @ppx.deriving(show)
+  type rec msg = SetTuplesAllowed(bool)
+
+  @ppx.deriving(show)
+  type rec intent = unit
+
+  let default = {
+    allowTuples: false,
+  }
+
+  let toSaved = (state: t) => {
+    open Json.Encode
+    object_(list{("allowTuples", bool(state.allowTuples))})
+  }
+  let fromSaved = (j: Js.Json.t) => {
+    open Json.Decode
+    {
       allowTuples: field("allowTuples", bool, j),
     }
   }
 
   let init = () => Tea.Cmd.none
 
-  let update = (state: t, msg: msg): (t, intent) =>
+  let update = (_state: t, msg: msg): (t, intent) =>
     switch msg {
-    | SetSidebarPanel(v) => ({...state, showSidebarDebuggerPanel: v}, ())
-    | SetTuplesAllowed(v) => ({...state, allowTuples: v}, ())
+    | SetTuplesAllowed(v) => ({allowTuples: v}, ())
     }
 }
 
 let title = "Contributing"
 
 @ppx.deriving(show)
-type rec t = {tunnelHost: TunnelHost.t, useAssets: UseAssets.t, general: General.t}
+type rec t = {
+  tunnelHost: TunnelHost.t,
```

Comment:
I guess this is what happens when the number of components start spiraling. This is fine, but I think we need to refactor something as we start to increase the number of options here